js/ui/appDisplay.js

```diff
@@ -620,12 +623,17 @@ var AllView = class AllView extends BaseAppView {
 
     loadGrid() {
         this._maybeAddAppCenterIcon();
+        this._maybeAddHackIcon();
         super.loadGrid();
     }
 
     getLayoutIds() {
         let layoutIds = super.getLayoutIds();
 
+        // Only show the hack icon if the clubhouse app is in the system
+        if (Clubhouse.getClubhouseApp())
+            layoutIds.splice(0, 0, HACK_APP_ID);
```

Comment:
nitpick: Maybe use `layout.unshift(HACK_APP_ID)` here, `splice()` tends to be confusing

Reply:
I've used unshift as you suggested.